core/src/data/clientGeoJsonSource.cpp

```diff
@@ -73,38 +78,28 @@ void ClientGeoJsonSource::addPoint(double* _coords) {
 
 }
 
-void ClientGeoJsonSource::addLine(double* _coords, int _lineLength) {
-    
-    std::vector<geojsonvt::LonLat> line(_lineLength, { 0, 0 });
-    for (int i = 0; i < _lineLength; i++) {
-        line[i] = { _coords[2 * i], _coords[2 * i + 1] };
-    }
-    
+void ClientGeoJsonSource::addLine(const Properties& _tags, const Coordinates& _line) {
+    auto& line = reinterpret_cast<const std::vector<geojsonvt::LonLat>&>(_line);
```

Comment:
Why are you doing a `const` with `reinterpret_cast`. I remember reading, its not recommended to do `const` with `reinterpret_cast`, but it might be for a specific use case. Just mentioning if you had checked for its usage!